src/sentry/monitors/serializers.py

```diff
@@ -35,7 +56,10 @@ def get_attrs(
         environments = {env.id: env for env in Environment.objects.filter(id__in=env_ids)}
 
         return {
-            monitor_env: {"environment": environments[monitor_env.environment_id]}
+            monitor_env: {
+                "environment": environments[monitor_env.environment_id],
+                "open_broken_detection": serialize(monitor_env.open_broken_detection, user),
```

Comment:
This is going to cause a n+1 query like you mentioned. I think what you'll want to do here is use the `env_ids` list to make a bulk query on the `MonitorEnvBrokenDetection` table joined with the incident table. You can then put that in a `Mapping[MonitorId, serialized MonitorEnvBrokenDetection]`